crates/re_data_source/src/data_loader/loader_archetype.rs

```diff
@@ -0,0 +1,157 @@
+use re_log_types::{DataRow, EntityPath, RowId, TimePoint};
+
+use crate::{DataLoader, DataLoaderError, LoadedData};
+
+// ---
+
+/// Loads data from any supported file or in-memory contents as native [`re_types::Archetype`]s.
+pub struct ArchetypeLoader;
+
+impl DataLoader for ArchetypeLoader {
+    #[inline]
+    fn name(&self) -> String {
+        "rerun.data_loaders.Archetype".into()
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
+    fn load_from_file(
+        &self,
+        store_id: re_log_types::StoreId,
+        filepath: std::path::PathBuf,
+        tx: std::sync::mpsc::Sender<LoadedData>,
+    ) -> Result<(), crate::DataLoaderError> {
+        use anyhow::Context as _;
+
+        if filepath.is_dir() {
+            return Ok(()); // simply not interested
+        }
+
+        re_tracing::profile_function!(filepath.display().to_string());
+
+        let contents = std::fs::read(&filepath)
+            .with_context(|| format!("Failed to read file {filepath:?}"))?;
+        let contents = std::borrow::Cow::Owned(contents);
+
+        self.load_from_file_contents(store_id, filepath, contents, tx)
+    }
+
+    fn load_from_file_contents(
+        &self,
+        _store_id: re_log_types::StoreId,
+        filepath: std::path::PathBuf,
+        contents: std::borrow::Cow<'_, [u8]>,
+        tx: std::sync::mpsc::Sender<LoadedData>,
+    ) -> Result<(), crate::DataLoaderError> {
+        re_tracing::profile_function!(filepath.display().to_string());
+
+        let entity_path = EntityPath::from_file_path(&filepath);
+
+        let timepoint = TimePoint::timeless();
+        // TODO: If I log these times, the space views disappear and cannot be created manually
+        // if let Ok(metadata) = filepath.metadata() {
+        //     use re_log_types::{Time, Timeline};
+        //
+        //     if let Some(created) = metadata.created().ok().and_then(|t| Time::try_from(t).ok()) {
+        //         timepoint.insert(Timeline::new_temporal("created_at"), created.into());
+        //     }
+        //     if let Some(modified) = metadata
+        //         .modified()
+        //         .ok()
+        //         .and_then(|t| Time::try_from(t).ok())
+        //     {
+        //         timepoint.insert(Timeline::new_temporal("modified_at"), modified.into());
+        //     }
+        //     if let Some(accessed) = metadata
+        //         .accessed()
+        //         .ok()
+        //         .and_then(|t| Time::try_from(t).ok())
+        //     {
+        //         timepoint.insert(Timeline::new_temporal("accessed_at"), accessed.into());
+        //     }
+        // }
```

Comment:
Instead of commenting out the code, use `if false` - that way the code won't rot, and it will be easy to turn back on when someone wants to test it